plugins/retry/index.js

```diff
@@ -22,6 +26,7 @@ function assertDelayQueue(delay, exchange) {
     };
 }
 
+/** @this {ContextChannel} */
```

Comment:
Annotation of `this` would be one of the key additions, that could ensure safety.